UpgradingACA-Py.md

```diff
@@ -0,0 +1,88 @@
+
+- Collect (if any) the actions to be taken to get from the version recorded in
+secure storage to the current [version.py]
+- Execute the actions from oldest to newest.
+    - If the same action is collected more than once (e.g., "Resave the
+Connection Records" is defined for two different versions), perform the action
+only once.
+- Store the current ACA-Py version (from [version.py]) in the secure storage
+  database.
+
+## Forced Offline Upgrades
+
+In some cases, it may be necessary to do an offline upgrade, where ACA-Py is
+taken off line temporarily, the database upgraded explicitly, and then
+ACA-Py re-deployed as normal. As yet, we do not know of any cases of this, but
+those deploying ACA-Py should be aware of this possibility. For example,
+we may at some point need an upgrade that **MUST NOT** be executed by more
+than one ACA-Py instance. In that case, a "normal" upgrade could be dangerous
+for deployments on container orchestration platforms like Kubernetes.
+
+If the Maintainers of ACA-Py recognize a case where ACA-Py must be upgraded while
+offline, a new Upgrade feature will be added that will prevent the "auto upgrade"
+process from executing. See [Issue XXXX] for the status of that feature.
+
+Those deploying ACA-Py upgrades for production installations should check in
+each [CHANGELOG.md] release entry about what upgrades (if any) will be run when
+upgrading to that version, and consider how they want those upgrades to run in
+their ACA-Py installation. In most cases, simply deploying the new version should
+be OK. If the number of records to be upgraded is high (such as a "resave
+connections" upgrade to a deployment with many, many connections), you may want
+to do a test upgrade offline first, to see if there is likely to be a service
+disruption during the upgrade. Plan accordingly!
+
+## Exceptions
+
+There are a couple of upgrade exception conditions to consider, as outlined
+in the following sections.
+
+### No version in secure storage
+
+Versions prior to ACA-Py 0.8.1 did not automatically populate the secure storage
+"version" record. That only occurred if an upgrade was explicitly executed. As
+of ACA-Py 0.8.1, the version record is added immediately after the secure
+storage database is created. If you are upgrading to ACA-Py 0.8.1 or later, and
+there is no version record in the secure storage, ACA-Py will assume you are
+running version 0.7.5, and execute the upgrade from that version to the current
+one. The choice of 0.7.5 as the default is safe because the same upgrades will
+be run of any version 0.7.5 and earlier, as can be seen in the [Upgrade
+Definition YML file]. Thus, even if you are really upgrading from (for example)
+0.6.2, the same upgrades are needed as from 0.7.5.
+
+### Forcing an upgrade
+
+If you would like to force an upgrade from a given version, a pair of ACA-Py
+configuration arguments can be used together. If you specify "`--from-version
```

Comment:
Maybe mention an example for clarity:
For instance, if the version in secure storage has been set as `v0.8.0` but for some reason you want to apply upgrades from `>=v0.7.2`, then you can specify "`--from-version v0.7.2`" and "`--force-upgrade`".

Reply:
Done.  Confirming @shaangill025 that the "v" is needed in "--from-version".  I've put it in, but is that right?

Reply:
Yes, as we use "v" in config too.